doc/source/data/random-access.rst

```diff
@@ -45,10 +45,24 @@ This means that each random lookup costs ~1 network RTT as well as a small amoun
 Performance
 -----------
 
-Since actor communication goes directly from worker to worker in Ray, the throughput of a RandomAccessDataset scales linearly with the number of workers available. As a rough measure, a single worker can provide ~2k individual gets/s and serve ~10k records/s for multigets, and this scales linearly as you increase the number of clients and workers for a single RandomAccessDataset. You will generally want more workers than clients, since the client does less computation than worker actors do.
+Since actor communication goes directly from worker to worker in Ray, the throughput of a RandomAccessDataset scales linearly with the number of workers available. As a rough measure, a single worker can provide ~2k individual gets/s and serve ~10k records/s for multigets, and this scales linearly as you increase the number of clients and workers for a single RandomAccessDataset. Large workloads may require hundreds of workers for sufficient throughput. You will also generally want more workers than clients, since the client does less computation than worker actors do.
 
 To debug performance problems, use ``random_access_ds.stats()``. This will return a string showing the actor-side measured latencies as well as the distribution of data blocks and queries across the actors. Load imbalances can cause bottlenecks as certain actors receive more requests than others. Ensure that load is evenly distributed across the key space to avoid this.
 
+It is important to note that the client (Ray worker process) can also be a bottleneck. To scale past the throughput of a single client, use multiple tasks to gather the data, for example:
+
+.. code-block:: python
+
+    @ray.remote
+    def fetch(rmap, keys):
+        return rmap.multiget(keys)
+
+    # Split the list of keys we want to fetch into 10 pieces.
+    pieces = np.array_split(all_keys, 10)
+
+    # Fetch from the RandomAccessDataset in parallel using 10 remote tasks.
+    print(ray.get([fetch.remote(rmap, p) for p in pieces]))
```

Comment:
should we define rmap and all_keys here to be self-contained code snippet.

Reply:
We should in the future. it's not needed for this low-hanging fruit improvement.